server/aesdsocket.c

```diff
@@ -0,0 +1,370 @@
+		    break;
+		}
+    		//syslog(LOG_INFO, "socket creation sucess \n");
+		
+		if (setsockopt(socket_fd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(int)) == -1) 
+		{
+		    syslog(LOG_ERR,"setsockopt failed \n");
+		    break;
+		}
+    		//syslog(LOG_INFO, "setsockopt sucess \n");
+    		
+		if (bind(socket_fd, p->ai_addr, p->ai_addrlen) == -1) 
+		{
+		    close(socket_fd);
+		    syslog(LOG_ERR,"server bind failed \n");
+		    break;
+		}
+		
+    	}
+    	
+    	/* STEP 3:
+    		free malloc for addr struct
+    		freeaddrinfo()
+    	*/
+    	freeaddrinfo(servinfo);
+    	
+    	/* STEP 4:
+    		if deamon specified, initialise it
+    		fork->create a new sid->change directory to root-> close and redirect stdin/err/out to /dev/null
+    	*/
+    	if(daemon_flag)
+    	{
+    		 pid = fork();
+    		
+    		if(pid <0)
+    		{
+    			syslog(LOG_ERR,"deamon fork failed \n");
+    			exit (EXIT_FAILURE);
+    		}
+    		else if (pid>0)
+    		{
+    			// this is the parent, terminate this
+    			syslog(LOG_INFO,"Parent terminated\n");
+			exit(0);
+    		}
+    		    		
+    		umask(0);
+    		//create a new session
+    		pid_t sid = setsid();
+    		if( sid <0)
+    		{
+    			syslog(LOG_ERR,"setsid failed");
+    			exit(EXIT_FAILURE);
+    		}
+    		
+    		// change dir to root
+    		chdir("/");
+    		
+    		//close fds
+		close(STDIN_FILENO);
+		close(STDOUT_FILENO);
+		close(STDERR_FILENO);
+		
+		//redirect to dev null
+		open("/dev/null", O_RDWR);
+		   			
+    	}
+    	
+    	/* STEP 5:
+    		Listens for and accepts a connection
+    	*/
+    	status = listen(socket_fd, BACKLOG);// how many pending connections queue will hold
+    	if (status == -1)
+    	{
+    		syslog(LOG_ERR, "listen failed \n");
+    		break;
+    	}
+    	
+    	while(1)
+    	{
+    		client_fd = accept(socket_fd, (struct sockaddr *)&client_address, &client_addrlen);
+    		if (client_fd ==-1)
+    		{
+    			syslog(LOG_ERR," accept failed \n");
+    			continue;
+    		}
+    		
+    		/*STEP 6:
+    			Logs message to the syslog “Accepted connection from xxx” 
+    		*/
+    		inet_ntop(client_address.ss_family, get_in_addr((struct sockaddr *)&client_address),
+            s, sizeof s);  // s is the ip address
+            	syslog(LOG_INFO,"Accepted connection from %s",s);
+            	
+            	
+	/* STEP 7: 
+		Receives data over the connection and appends to file /var/tmp/aesdsocketdata, creating this file if it doesn’t exist.
+	*/
+            
+		
+            //move to the start of the file
+            off_t seekset_to_beg = lseek(logfile_fd, 0, SEEK_SET);
+	    if(seekset_to_beg == -1)
+		{
+			syslog(LOG_ERR,"lseek failed \n");
+			break;
+		}
+
+            
+            do
+            {
+            
+		recv_bytes = recv(client_fd, buf, BUFFER_LEN, 0);
+		if (recv_bytes == -1)
+		{
+			syslog (LOG_ERR, "receving of bytes failed \n");
+			break;
+		}
+		
+		//wrie to log file
+		status = write (logfile_fd, buf, recv_bytes);
```

Comment:
@ishassharmaa with this design you will write to the file before the packet completes, which may cause problems with future assignments.